### DISCUSSION TITLE
6.4.2 프론트엔드 배포 질문드려요

### DISCUSSION THREAD

**Discussion by iknocho** [fsoftwareengineer/todo-application#87] [질문과 답변] [answered]
![스크린샷 2022-02-22 오후 10 36 42](https://user-images.githubusercontent.com/49403520/155143260-b42fa178-3104-4ed5-bcd6-d4f0ef0d64c2.png)
엘라스틱빈스톡 환경까지 만들고 프론트엔드 배포를 완료했습니다. 정확히는 모르겠지만 빌드한 파일이 올라가지 않고  public 디렉토리의 index.html 파일이 EC2에 올라간 것 같습니다.
![스크린샷 2022-02-22 오후 9 45 48](https://user-images.githubusercontent.com/49403520/155143761-f48b4976-3abd-4eb7-9a93-1d5e197c8b49.png)
![스크린샷 2022-02-22 오후 10 34 20](https://user-images.githubusercontent.com/49403520/155143832-7e568b7c-dd19-46da-a6d1-202286c7d8d6.png)
디펜던시를 수정해서 빌드하였는데도 @material 오류도 계속 발생하고 있습니다.
로컬에서 빌드해서 serve를 통해서 배포하였을 경우에는 아무런 문제없이 잘 실행됩니다.
![스크린샷 2022-02-22 오후 10 43 24](https://user-images.githubusercontent.com/49403520/155144513-6bac4d95-4c78-4bcc-af28-b4ef9aaf8f7a.png)
혹시 이 문제 해결방법 알 수 있을까요?
엘라스틱 빈스톡 환경만들때
instance deployment: you didn't specify a node.js version in the 'package.json' file in your source bundle. the deployment didn't install a specific node.js version.
이러한 오류가 있긴하였는데 이 문제 때문인지도 궁금합니다.

**Comment by fsoftwareengineer** [comment, marked as answer]
eb로 초기화 시 git을 사용하셨다면 commit을 하고 eb deploy해야합니다. git으로 소스를 관리하고 있는지 확인 해 주시기 바랍니다.

**Reply by iknocho** [reply]
밑에 저랑 비슷한 질문을 보고 commit 후 deploy 와 .git 파일을 삭제 후 deploy 결과 문제를 해결하지 못했습니다.
![스크린샷 2022-02-23 오전 11 06 49](https://user-images.githubusercontent.com/49403520/155251948-149fc120-27f5-45fb-92a5-7fb62c8b1c29.png)
웹 소켓 오류가 발생하고 있다고 하는데 이것 때문일까요?
좌측 화면 로컬에서 돌리는 코드와 우측의 AWS올릴 코드가 다른 것을 확인할 수 있었습니다.

**Comment by fsoftwareengineer** [comment]
package.json을 깃허브 소스와 비교해보면 어떤가요? dev에만 추가되고 프로덕션에서는 설치가 안된게 아닌지 확인 해 보는 것이 좋을 것 같습니다.

**Reply by iknocho** [reply]
감사합니다. 해결했어요!!! 깃허브에 올린 폴더 완전히 삭제 후 새로운 폴더에서 진행하여 해결했습니다. 제가 깃허브에 올릴때 
프로젝트 디렉토리 위에 디렉토리를 하나 더 만들어서 올린게 문제가 된 것 같습니다.
